core/Pos/Crypto/SafeSigning.hs

```diff
@@ -139,7 +168,7 @@ safeDeterministicKeyGen
     -> PassPhrase
     -> (PublicKey, EncryptedSecretKey)
 safeDeterministicKeyGen seed pp =
-    bimap PublicKey (mkEncSecret pp) (safeCreateKeypairFromSeed seed pp)
+    bimap PublicKey (mkEncSecretWithSalt (S.mkSalt seed) pp) (safeCreateKeypairFromSeed seed pp)
```

Comment:
So if the attacker knows the salt, they know the seed ⇒ they know the keypair. That seems really bad.

Reply:
Perhaps you can take a `(ByteString, ByteString)` instead.

Reply:
Definitiely, add one hash round.

Reply:
A hash will probably do too, yeah. `bimap PublicKey (mkEncSecretWithSalt (S.mkSalt (hash seed)) pp) (safeCreateKeypairFromSeed seed pp)` (it's important that we use the hash as a salt, not as a seed)

Reply:
Mad bug